The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/12562188117/job/35022288955) [(plain)](https://github.com/rust-lang/rust/commit/dc9e8045acc8c6a3d76a89d62eeee81e27bcf453/checks/35022288955/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---
#13 2.917 Building wheels for collected packages: reuse
#13 2.918   Building wheel for reuse (pyproject.toml): started
#13 3.165   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.166   Created wheel for reuse: filename=reuse-4.0.3-cp310-cp310-manylinux_2_35_x86_64.whl size=132720 sha256=026f3bb0f1aa8090b861fd0a0939cb1a782396d84c8aab7875096557d637a0f6
#13 3.166   Stored in directory: /tmp/pip-ephem-wheel-cache-yjirn3my/wheels/3d/8d/0a/e0fc6aba4494b28a967ab5eaf951c121d9c677958714e34532
#13 3.169 Installing collected packages: boolean-py, binaryornot, tomlkit, reuse, python-debian, markupsafe, license-expression, jinja2, chardet, attrs
#13 3.570 Successfully installed attrs-23.2.0 binaryornot-0.4.4 boolean-py-4.0 chardet-5.2.0 jinja2-3.1.4 license-expression-30.3.0 markupsafe-2.1.5 python-debian-0.1.49 reuse-4.0.3 tomlkit-0.13.0
#13 3.570 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 4.107 Collecting virtualenv
#13 4.107 Collecting virtualenv
#13 4.157   Downloading virtualenv-20.28.0-py3-none-any.whl (4.3 MB)
#13 4.357      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 21.7 MB/s eta 0:00:00
#13 4.416 Collecting platformdirs<5,>=3.9.1
#13 4.423   Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
#13 4.463 Collecting filelock<4,>=3.12.2
#13 4.493 Collecting distlib<1,>=0.3.7
#13 4.500   Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
#13 4.514      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 39.0 MB/s eta 0:00:00
#13 4.514      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 469.0/469.0 KB 39.0 MB/s eta 0:00:00
#13 4.596 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 4.780 Successfully installed distlib-0.3.9 filelock-3.16.1 platformdirs-4.3.6 virtualenv-20.28.0
#13 DONE 4.9s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      118720 kB
DirectMap2M:     6172672 kB
DirectMap1G:    12582912 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint,cpp:fmt
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
WARN: currently no CI rustc builds have rustc debug assertions enabled. Please either set `rust.debug-assertions` to `false` if you want to use download CI rustc or set `rust.download-rustc` to `false`.
downloading https://static.rust-lang.org/dist/2024-11-27/rustfmt-nightly-x86_64-unknown-linux-gnu.tar.xz
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 31.77s
##[endgroup]
fmt check
Diff in /checkout/library/core/src/slice/mod.rs:3193:
         sort::select::partition_at_index(self, index, |a: &T, b: &T| compare(a, b) == Less)
 
-
     /// Reorders the slice with a key extraction function such that the element at `index` is at a
     /// Reorders the slice with a key extraction function such that the element at `index` is at a
     /// sort-order position. All elements before `index` will have keys `<=` the key at `index`, and
     /// all elements after will have keys `>=`.
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/core/src/panic/panic_info.rs" "/checkout/library/core/src/panic/location.rs" "/checkout/library/core/src/panic.rs" "/checkout/library/core/src/option.rs" "/checkout/tests/run-make/alloc-no-rc/rmake.rs" "/checkout/library/core/src/sync/atomic.rs" "/checkout/library/core/src/sync/exclusive.rs" "/checkout/library/core/src/sync/mod.rs" "/checkout/library/core/src/ascii.rs" "/checkout/library/core/src/asserting.rs" "/checkout/library/core/src/primitive_docs.rs" "/checkout/tests/run-make/wasm-spurious-import/rmake.rs" "/checkout/tests/run-make/wasm-spurious-import/main.rs" "/checkout/library/core/src/prelude/common.rs" "/checkout/library/core/src/prelude/mod.rs" "/checkout/library/core/src/lib.miri.rs" "/checkout/library/core/src/pat.rs" "/checkout/library/core/src/unsafe_binder.rs" "/checkout/library/core/src/cell.rs" "/checkout/tests/run-make/c-static-dylib/foo.rs" "/checkout/tests/run-make/c-static-dylib/bar.rs" "/checkout/tests/run-make/c-static-dylib/rmake.rs" "/checkout/tests/run-make/separate-link-fail/foo.rs" "/checkout/tests/run-make/separate-link-fail/rmake.rs" "/checkout/library/core/src/slice/iter/macros.rs" "/checkout/library/core/src/slice/rotate.rs" "/checkout/library/core/src/slice/ascii.rs" "/checkout/library/core/src/slice/index.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/rmake.rs" "/checkout/library/core/src/slice/raw.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/src/lib.rs" "/checkout/tests/run-make/rustdoc-scrape-examples-test/examples/ex.rs" "/checkout/library/core/src/slice/sort/unstable/quicksort.rs" "/checkout/library/core/src/slice/sort/unstable/heapsort.rs" "/checkout/library/core/src/slice/sort/unstable/mod.rs" "/checkout/library/core/src/slice/sort/shared/pivot.rs" "/checkout/library/core/src/slice/sort/shared/smallsort.rs" "/checkout/library/core/src/slice/sort/shared/mod.rs" "/checkout/library/core/src/slice/sort/mod.rs" "/checkout/tests/run-make/mixing-formats/foo.rs" "/checkout/tests/run-make/mixing-formats/bar2.rs" "/checkout/tests/run-make/mixing-formats/baz2.rs" "/checkout/tests/run-make/mixing-formats/rmake.rs" "/checkout/tests/run-make/mixing-formats/bar1.rs" "/checkout/tests/run-make/mixing-formats/baz.rs" "/checkout/library/core/src/slice/sort/stable/tiny.rs" "/checkout/library/core/src/slice/sort/stable/quicksort.rs" "/checkout/library/core/src/slice/sort/stable/merge.rs" "/checkout/library/core/src/slice/sort/stable/drift.rs" "/checkout/library/core/src/slice/sort/stable/mod.rs" "/checkout/library/core/src/slice/sort/select.rs" "/checkout/library/core/src/slice/mod.rs" "/checkout/library/core/src/slice/memchr.rs" "/checkout/library/core/src/slice/iter.rs" "/checkout/tests/run-make/extern-multiple-copies2/bar.rs" "/checkout/library/core/src/slice/cmp.rs" "/checkout/tests/run-make/extern-multiple-copies2/foo1.rs" "/checkout/library/core/src/slice/specialize.rs" "/checkout/tests/run-make/extern-multiple-copies2/rmake.rs" "/checkout/tests/run-make/extern-multiple-copies2/foo2.rs" "/checkout/tests/run-make/invalid-symlink-search-path/rmake.rs" "/checkout/library/core/src/char/decode.rs" "/checkout/library/core/src/char/methods.rs" "/checkout/tests/run-make/print-target-cpus-native/rmake.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Dec 31 17:25:34 UTC 2024
  network time: Tue, 31 Dec 2024 17:25:34 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/slice/mod.rs
+++ b/library/core/src/slice/mod.rs
@@ -3069,19 +3069,18 @@ impl<T> [T] {
         sort::unstable::sort(self, &mut |a, b| f(a).lt(&f(b)));
     }
 
-    /// Reorders the slice such that the element at `index` after the reordering is at its final
-    /// sorted position.
+    /// Reorders the slice such that the element at `index` is at a sort-order position. All
+    /// elements before `index` will then be `<=` this value, and all elements after will be `>=`.
     ///
-    /// This reordering has the additional property that any value at position `i < index` will be
-    /// less than or equal to any value at a position `j > index`. Additionally, this reordering is
-    /// unstable (i.e. any number of equal elements may end up at position `index`), in-place (i.e.
-    /// does not allocate), and runs in *O*(*n*) time. This function is also known as "kth element"
-    /// in other libraries.
+    /// This reordering is unstable (i.e. any element that compares equal to the nth element may end
+    /// up at that position), in-place (i.e.  does not allocate), and runs in *O*(*n*) time. This
+    /// function is also known as "kth element" in other libraries.
     ///
-    /// It returns a triplet of the following from the reordered slice: the subslice prior to
-    /// `index`, the element at `index`, and the subslice after `index`; accordingly, the values in
-    /// those two subslices will respectively all be less-than-or-equal-to and
-    /// greater-than-or-equal-to the value of the element at `index`.
+    /// Returns a triple partitioning the reordered slice:
+    ///
+    /// * The unsorted subslice before `index` (elements all pass `x <= self[index]`)
+    /// * The element at `index`
+    /// * The unsorted subslice after `index` (elements all pass `x >= self[index]`)
     ///
     /// # Current implementation
     ///
@@ -3094,7 +3093,7 @@ impl<T> [T] {
     ///
     /// # Panics
     ///
-    /// Panics when `index >= len()`, meaning it always panics on empty slices.
+    /// Panics when `index >= len()`, and so always panics on empty slices.
     ///
     /// May panic if the implementation of [`Ord`] for `T` does not implement a [total order].
     ///
@@ -3103,8 +3102,7 @@ impl<T> [T] {
     /// ```
     /// let mut v = [-5i32, 4, 2, -3, 1];
     ///
-    /// // Find the items less than or equal to the median, the median, and greater than or equal to
-    /// // the median.
+    /// // Find the items `<=` the median, the median, and `>=` the median.
     /// let (lesser, median, greater) = v.select_nth_unstable(2);
     ///
     /// assert!(lesser == [-3, -5] || lesser == [-5, -3]);
@@ -3130,19 +3128,19 @@ impl<T> [T] {
         sort::select::partition_at_index(self, index, T::lt)
     }
 
-    /// Reorders the slice with a comparator function such that the element at `index` after the
-    /// reordering is at its final sorted position.
+    /// Reorders the slice with a comparator function such that the element at `index` is at a
+    /// sort-order position. All elements before `index` will then be `<=` this value, and all
+    /// elements after will be `>=` according to the comparator function.
     ///
-    /// This reordering has the additional property that any value at position `i < index` will be
-    /// less than or equal to any value at a position `j > index` using the comparator function.
-    /// Additionally, this reordering is unstable (i.e. any number of equal elements may end up at
-    /// position `index`), in-place (i.e. does not allocate), and runs in *O*(*n*) time. This
+    /// This reordering is unstable (i.e. any element that compares equal to the nth element may end
+    /// up at that position), in-place (i.e.  does not allocate), and runs in *O*(*n*) time. This
     /// function is also known as "kth element" in other libraries.
     ///
-    /// It returns a triplet of the following from the slice reordered according to the provided
-    /// comparator function: the subslice prior to `index`, the element at `index`, and the subslice
-    /// after `index`; accordingly, the values in those two subslices will respectively all be
-    /// less-than-or-equal-to and greater-than-or-equal-to the value of the element at `index`.
+    /// Returns a triple partitioning the reordered slice:
+    ///
+    /// * The unsorted subslice before `index` (elements all pass `compare(x, self[index]).is_le()`)
+    /// * The element at `index`
+    /// * The unsorted subslice after `index` (elements all pass `compare(x, self[index]).is_ge()`)
     ///
     /// # Current implementation
     ///
@@ -3155,7 +3153,7 @@ impl<T> [T] {
     ///
     /// # Panics
     ///
-    /// Panics when `index >= len()`, meaning it always panics on empty slices.
+    /// Panics when `index >= len()`, and so always panics on empty slices.
     ///
     /// May panic if `compare` does not implement a [total order].
     ///
@@ -3164,13 +3162,13 @@ impl<T> [T] {
     /// ```
     /// let mut v = [-5i32, 4, 2, -3, 1];
     ///
-    /// // Find the items less than or equal to the median, the median, and greater than or equal to
-    /// // the median as if the slice were sorted in descending order.
-    /// let (lesser, median, greater) = v.select_nth_unstable_by(2, |a, b| b.cmp(a));
+    /// // Find the items `>=` the median, the median, and `<=` the median, by using a reversed
+    /// // comparator.
+    /// let (before, median, after) = v.select_nth_unstable_by(2, |a, b| b.cmp(a));
     ///
-    /// assert!(lesser == [4, 2] || lesser == [2, 4]);
+    /// assert!(before == [4, 2] || before == [2, 4]);
     /// assert_eq!(median, &mut 1);
-    /// assert!(greater == [-3, -5] || greater == [-5, -3]);
+    /// assert!(after == [-3, -5] || after == [-5, -3]);
     ///
     /// // We are only guaranteed the slice will be one of the following, based on the way we sort
     /// // about the specified index.
@@ -3195,19 +3193,19 @@ impl<T> [T] {
         sort::select::partition_at_index(self, index, |a: &T, b: &T| compare(a, b) == Less)
     }
 
-    /// Reorders the slice with a key extraction function such that the element at `index` after the
-    /// reordering is at its final sorted position.
+    /// Reorders the slice with a key extraction function such that the element at `index` is at a
+    /// sort-order position. All elements before `index` will have keys `<=` the key at `index`, and
+    /// all elements after will have keys `>=`.
     ///
-    /// This reordering has the additional property that any value at position `i < index` will be
-    /// less than or equal to any value at a position `j > index` using the key extraction function.
-    /// Additionally, this reordering is unstable (i.e. any number of equal elements may end up at
-    /// position `index`), in-place (i.e. does not allocate), and runs in *O*(*n*) time. This
+    /// This reordering is unstable (i.e. any element that compares equal to the nth element may end
+    /// up at that position), in-place (i.e.  does not allocate), and runs in *O*(*n*) time. This
     /// function is also known as "kth element" in other libraries.
     ///
-    /// It returns a triplet of the following from the slice reordered according to the provided key
-    /// extraction function: the subslice prior to `index`, the element at `index`, and the subslice
-    /// after `index`; accordingly, the values in those two subslices will respectively all be
-    /// less-than-or-equal-to and greater-than-or-equal-to the value of the element at `index`.
+    /// Returns a triple partitioning the reordered slice:
+    ///
+    /// * The unsorted subslice before `index` (elements all pass `f(x) <= f(self[index])`)
+    /// * The element at `index`
+    /// * The unsorted subslice after `index` (elements all pass `f(x) >= f(self[index])`)
     ///
     /// # Current implementation
     ///
@@ -3229,8 +3227,8 @@ impl<T> [T] {
     /// ```
     /// let mut v = [-5i32, 4, 1, -3, 2];
     ///
-    /// // Find the items less than or equal to the median, the median, and greater than or equal to
-    /// // the median as if the slice were sorted according to absolute value.
+    /// // Find the items <= the median absolute value, the median absolute value, and >= the median
+    /// // absolute value.
     /// let (lesser, median, greater) = v.select_nth_unstable_by_key(2, |a| a.abs());
     ///
     /// assert!(lesser == [1, 2] || lesser == [2, 1]);

